keras/backend/tensorflow_backend.py

```diff
@@ -1924,3 +1925,24 @@ def ctc_decode(y_pred, input_length, greedy=True, beam_width=100,
                      for st in decoded]
 
     return (decoded_dense, log_prob)
+
+
+class ReverseGradient(object):
```

Comment:
I can clarify this a bit since I wrote the original [here](https://github.com/pumpikano/tf-dann/blob/master/flip_gradient.py). The grad reversal layer overrides the gradient of identity with an expression that has a hyperparam lambda. Since different instances can have different lambdas, we need to register a new gradient  op for each instance of the grad reversal layer. This is accomplished by assigning each new gradient op a unique name via `num_calls`. The only reason this is a callable class is to avoid `num_calls` being a global.

A couple comments on the implementation here:
1. It doesn't make sense for the lambda hyperparam to be an arg to init here - it defeats the entire purpose of implementing this as a class.
2. The only reason this is a class is to organize `num_calls`, so the class should be instantiated once right here to make a op function and the class itself need not be exported from the module, e.g. https://github.com/pumpikano/tf-dann/blob/master/flip_gradient.py#L22. Alternately, `num_calls` can be a global or captured in a closure - this is a style decision that is up to @fchollet.

Hope this helps.